### PR TITLE
[PL-51498] SMP 0.17.0 Release Notes

### PR DESCRIPTION
**Pull request by saifeemustafaqharness**
[PL-51498](https://harness.atlassian.net/browse/PL-51498)

https://harness.atlassian.net/wiki/spaces/PD/pages/21808939009/Release+Notes+tickets+for+v0.17.0+release

## Description

* Please describe your changes: SMP Release Notes
* Jira/GitHub Issue numbers (if any): [PL-51498](https://harness.atlassian.net/browse/PL-51498)
* Preview links/images (Internal contributors only): [Self-Managed Enterprise Edition](https://665a41828484c0ab6ce4531b--harness-developer.netlify.app/release-notes/self-managed-enterprise-edition)

|          |           |            |           |           |
|----------|-----------|------------|-----------|-----------|
| PL-50453 | FFM-11292 | STO-7350   | CDS-95967 | CDS-94707 |
| PL-50438 | FFM-11253 | STO-7287   | CDS-95845 | CDS-94620 |
| PL-49256 | FFM-11231 | STO-7029   | CDS-95843 | CDS-94530 |
| PL-48980 | FFM-11230 | PIPE-18652 | CDS-95840 | CDS-94460 |
| PL-48948 | FFM-11215 | PIPE-16923 | CDS-95726 | CDS-94376 |
| PL-48891 | FFM-11213 | PIPE-16893 | CDS-95527 | CDS-94368 |
| PL-48854 | FFM-11199 | PIPE-16866 | CDS-95526 | CDS-94366 |
| PL-48788 | FFM-11157 | PIPE-15903 | CDS-95485 | CDS-94355 |
| PL-48781 | FFM-11125 | IDP-2680   | CDS-95481 | CDS-94272 |
| PL-48671 | FFM-11114 | CCM-17267  | CDS-95365 | CDS-94245 |
| PL-48606 | FFM-11104 | CCM-15955  | CDS-95312 | CDS-94113 |
| PL-48532 | CI-12038  |            | CDS-95273 | CDS-94092 |
| PL-48488 | CI-11995  |            | CDS-95250 | CDS-94084 |
| PL-48420 | CI-11953  |            | CDS-95241 | CDS-94059 |
| PL-47940 | CI-11891  |            | CDS-95219 | CDS-93931 |
| PL-47914 | CI-11882  |            | CDS-95209 | CDS-93678 |
| PL-47753 | CI-11819  |            | CDS-95164 | CDS-92836 |
| PL-47632 | CI-11792  |            | CDS-95136 | CDS-92633 |
| PL-47614 | CI-11709  |            | CDS-95061 | CDS-92631 |
| PL-43907 | CI-11672  |            | CDS-95048 | CDS-91534 |
| PL-43094 | CI-11556  |            | CDS-94962 | CDS-91348 |
| PL-40727 | CI-11305  |            | CDS-94937 | CDS-91077 |
| PL-38735 |           |            | CDS-94886 | CDS-85185 |
|          |           |            | CDS-94801 |           |
|          |           |            |           |           |

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

[PL-51498]: https://harness.atlassian.net/browse/PL-51498?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PL-51498]: https://harness.atlassian.net/browse/PL-51498?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by batraprashant — DISMISSED** [review]

File: release-notes/self-managed-enterprise-edition.md [hunk 1]
```diff
@@ -82,6 +82,128 @@ If you don't use Helm to upgrade Harness Self-Managed Enterprise Edition, follow
...
+| CI Manager | 1.21.5 |
+| Pipeline Service | 1.68.2 |
+| Platform Service | 1.17.1 |
+| Access Control Service | 1.39.1 |
+| Delegate | 24.03.82600 |
+| Change Data Capture | 1.5.4 |
+| STO Core | 1.90.1 |
+| Test Intelligence Service | 1.13.1 |
+| NG UI | 1.14.5 |
+| LE NG | 1.1.0 |
```
Comment (brian-f-harness) [inline release-notes/self-managed-enterprise-edition.md:103]: Artifact versions for 0.17.0 are available here: https://harness.atlassian.net/wiki/spaces/PD/pages/21785346073/SMP+Release+Sign-off+May+2024+build+0.17.0#Artifact-Versions%3A

File: release-notes/self-managed-enterprise-edition.md [hunk 2]
```diff
@@ -82,6 +82,198 @@ If you don't use Helm to upgrade Harness Self-Managed Enterprise Edition, follow
...
+
+- 2FA reset emails failed to display the QR code properly due to the recent deprecation of Google APIs. The method for generating QR codes has been updated, resolving the issue and ensuring QR codes are now correctly included in 2FA reset emails. This item requires Harness Delegate version 24.04.82804. For information about Harness Delegate features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate) (PL-48980, ZD-61314, ZD-61420, ZD-61486)
+
+- The link to Slack on the UI Help section was not functioning. The community link has been updated to ensure it works correctly. (PL-48948, ZD-61203)
+
+- Delegates with mTLS enabled were able to send a heartbeat to Harness Manager despite being configured with a non-agent endpoint. Resolved this by ensuring the `isNg` flag is correctly propagated when delegates send heartbeats to Harness Manager. (PL-48891, ZD-60974) This item requires Harness Delegate version 24.04.82901. For information about Harness Delegate features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (PL-48891, ZD-60974)
```
Comment (brian-f-harness) [inline release-notes/self-managed-enterprise-edition.md:223]:
```suggestion
- Delegates with mTLS enabled were able to send a heartbeat to Harness Manager despite being configured with a non-agent endpoint. Resolved this by ensuring the `isNg` flag is correctly propagated when delegates send heartbeats to Harness Manager. This item requires Harness Delegate version 24.04.82901. For information about Harness Delegate features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (PL-48891, ZD-60974)
```

File: release-notes/self-managed-enterprise-edition.md [hunk 3]
```diff
@@ -83,6 +83,200 @@ If you don't use Helm to upgrade Harness Self-Managed Enterprise Edition, follow
...
+- This release includes new default settings that enable you to configure s3-compatible caching in self-hosted build infrastructures. You can configure the endpoint URL, region, bucket name, access key, and secret key. These options are behind the feature flags CI_ENABLE_DLC_SELF_HOSTED (for Docker layer caching) and CI_ENABLE_CACHE_INTEL_SELF_HOSTED (for Cache Intelligence). Contact [Harness Support](mailto:support@harness.io) to enable them. (CI-11953)
+
+- If you need to allow empty environment variables in your CI pipelines, you can enable the feature flag `CI_USE_LESS_STRICT_EVALUATION_FOR_MAP_VARS` by contacting [Harness Support](mailto:support@harness.io). (CI-11882, CI-11305, CI-11672, ZD-57626)
+
+### Breaking changes
+
+#### Continuous Delivery
+
+- The RepoName, FilePath, and ConnectorRef parameters are marked as required in the Git import APIs for pipelines, templates, input sets, services, environments, infrastructure definitions, and service overrides. These parameters were optional before but made mandatory now as the APIs require these to work. (CDS-94245)
+
```
Comment (brian-f-harness) [inline release-notes/self-managed-enterprise-edition.md:279]: Recommend moving above new features

File: release-notes/self-managed-enterprise-edition.md [hunk 4]
```diff
@@ -83,6 +83,200 @@ If you don't use Helm to upgrade Harness Self-Managed Enterprise Edition, follow
...
+### Early access feature
+
+#### Continuous Delivery
+
+- Harness UI now supports  pipeline rollback failure strategy in CI, STO, Approval, Custom, FF stage, step, and step-group. This functionality is behind the feature flag, `CDS_ALLOW_EXPRESSION_RESOLUTION_PIPELINE_ROLLBACK`. To enable a feature flag in your Harness account, contact [Harness Support](mailto:support@harness.io). (PIPE-15903)
+
+#### Continuous Integration
+
+- This release includes new default settings that enable you to configure s3-compatible caching in self-hosted build infrastructures. You can configure the endpoint URL, region, bucket name, access key, and secret key. These options are behind the feature flags CI_ENABLE_DLC_SELF_HOSTED (for Docker layer caching) and CI_ENABLE_CACHE_INTEL_SELF_HOSTED (for Cache Intelligence). Contact [Harness Support](mailto:support@harness.io) to enable them. (CI-11953)
```
Comment (brian-f-harness) [inline release-notes/self-managed-enterprise-edition.md:270]:
```suggestion
- This release includes new default settings that enable you to configure s3-compatible caching in self-hosted build infrastructures. You can configure the endpoint URL, region, bucket name, access key, and secret key. These options are behind the feature flags `CI_ENABLE_DLC_SELF_HOSTED` (for Docker layer caching) and `CI_ENABLE_CACHE_INTEL_SELF_HOSTED` (for Cache Intelligence). Contact [Harness Support](mailto:support@harness.io) to enable them. (CI-11953)
```

File: release-notes/self-managed-enterprise-edition.md (hunk 3 above)
Comment (brian-f-harness) [reply release-notes/self-managed-enterprise-edition.md]: @saifeemustafaqharness I moved these items to the top of the list. Early access and breaking changes should be first.

File: release-notes/self-managed-enterprise-edition.md [hunk 5]
```diff
@@ -83,6 +83,245 @@ If you don't use Helm to upgrade Harness Self-Managed Enterprise Edition, follow
...
+#### Alternative air gap bundle download method
+
+Some admins might not have Google account access to download air gap bundles. As an alternative, you can use `gsutil`. For `gsutil` installation instructions, go to [Install gsutil](https://cloud.google.com/storage/docs/gsutil_install) in the Google Cloud documentation.
+
+```
+gsutil -m cp \
+  "gs://smp-airgap-bundles/harness-0.17.0/ccm_images.tgz" \
+  "gs://smp-airgap-bundles/harness-0.17.0/cdng_images.tgz" \
+  "gs://smp-airgap-bundles/harness-0.17.0/ce_images.tgz" \
+  "gs://smp-airgap-bundles/harness-0.17.0/cet_images.tgz" \
```
Comment (M-Tsur) [inline release-notes/self-managed-enterprise-edition.md:157]: CET was removed in this version, so I think we should remove the image link from the command. Maybe we should add some note in some other place to notify about it

Comment (M-Tsur) [reply release-notes/self-managed-enterprise-edition.md:157]: You can check this [slack thread](https://harness.slack.com/archives/C0501BCG1EF/p1714621727017519) for reference

Comment (brian-f-harness) [reply release-notes/self-managed-enterprise-edition.md:157]: Nice catch @M-Tsur. Thanks.

Comment (brian-f-harness) [inline release-notes/self-managed-enterprise-edition.md:157]:
```suggestion
```

**Review by douglas-j-bothwell — APPROVED** [review]
A few minor sugs otherwise LGTM

File: release-notes/self-managed-enterprise-edition.md [hunk 6]
```diff
@@ -82,6 +82,128 @@ If you don't use Helm to upgrade Harness Self-Managed Enterprise Edition, follow
...
 :::
 
+## June xx, 2024, version 0.17.0
+
+This release includes the following Harness module and component versions.
+
+| **Name** | **Version** |
+| :-- | :--: |
+| Helm Chart | [0.17.0](https://github.com/harness/helm-charts/releases/tag/harness-0.17.0) |
+| Air Gap Bundle | [0.17.0](https://console.cloud.google.com/storage/browser/smp-airgap-bundles/harness-0.17.0) |
```
Comment (pratmit) [inline release-notes/self-managed-enterprise-edition.md:135]: Airgap bundle will be out in IST morning, either we can remove this for now or publish RN once they are available

**Review by pratmit — APPROVED** [review]
Airgap bundle will be out in IST morning, either we can remove this for now or publish RN once they are available
Otherwise LGTM

**Comment by brian-f-harness** [comment]
@bharatgoelharness Please review the Looker info. I'll merge this PR tomorrow after the air gap bundle is uploaded.
cc @pratmit

**Review by brian-f-harness — APPROVED** [review]

**Pull request merged**